### PR TITLE
Add code style documentation

### PR DESCRIPTION
**Pull request by rpanchyk**

### CODE REVIEW

File: docs/code-style.md [hunk 1]
```diff
@@ -1,6 +1,136 @@
...
+import java.util.*;
+
+
+// good
+import java.util.HashMap;
+import java.util.Map;
+```
+
+### Variable and method naming
+Preferred to use `camel-case` naming convention for variables and methods.
```
Comment (snahornyi) [inline docs/code-style.md:35]: Nitpick. `camelCase`

**Review by snahornyi — APPROVED** [review]

**Comment by And1sS** [comment]
What do you think about adding this section?

```
### Data retrieval calls of same result
Try not to retrieve same data more than once:

// bad
if (getData() != null) {
   final Data resolvedData = resolveData(getData());

//  good
final Data data = getData();
if (data != null) {
   final Data resolvedData = resolveData(data);
```

**Review by And1sS — DISMISSED** [review]

File: docs/code-style.md [hunk 2]
```diff
@@ -1,6 +1,237 @@
...
+
+The [pom.xml](../pom.xml) is configured to enforce a coding style defined in [checkstyle.xml](../checkstyle.xml) when
+maven `validate` phase executed.
+
+## Formatting
+
+The project uses formatting rules described in `.editorconfig` file. Most of the popular IDEs have support of it. For
+example, in `IntelliJ IDEA` hit `Code -> Reformat Code` to organize your code.
+
+## Code conventions
```
Comment (Net-burst) [inline docs/code-style.md:16]: I'd also add some general and simple things like:
- Prefer composition over inheritance.
- Methods should have only 1 return point (not enforced)

**Review by Net-burst — DISMISSED** [review]

File: docs/code-style.md (hunk 2 above)
Comment (rpanchyk) [reply docs/code-style.md:16]: i'd say it's more like design. let me skip them.

**Review by snahornyi — APPROVED** [review]

**Pull request merged**